src/node-lib/curriculum-api-2023/queries/lessonDownloads/lessonDownloads.query.ts

```diff
@@ -31,12 +46,72 @@ const lessonDownloadsQuery =
       });
     }
 
-    const page = pages[0];
-    const unit = units[0];
-    const nextLessons = getNextLessonsInUnit(unit?.lessons, args.lessonSlug);
+    const { download_assets, unit_lessons } = res;
+
+    const {
+      has_slide_deck_asset_object,
+      has_worksheet_asset_object,
+      has_supplementary_asset_object,
+      has_worksheet_answers_asset_object,
+      has_worksheet_google_drive_downloadable_version,
+      starter_quiz,
+      exit_quiz,
+      is_legacy,
+      expired,
+    } = downloadsAssetData.parse(download_assets[0]);
+
+    const currentLesson = unit_lessons.find(
+      (lesson) => lesson.lesson_slug === args.lessonSlug,
+    );
+
+    const copyright = keysToCamelCase(
+      currentLesson?.lesson_data.copyright_content,
+    );
+
+    const parsedCurrentLesson =
+      syntheticUnitvariantLessonsSchema.parse(currentLesson);
+
+    // parsing currentLesson returns empty copyright object?
+
+    const downloadsData = {
+      hasSlideDeckAssetObject: has_slide_deck_asset_object,
+      hasStarterQuiz: starter_quiz ? true : false,
+      hasExitQuiz: exit_quiz ? true : false,
+      hasWorksheetAssetObject: has_worksheet_asset_object,
+      hasWorksheetAnswersAssetObject: has_worksheet_answers_asset_object,
+      hasWorksheetGoogleDriveDownloadableVersion:
+        has_worksheet_google_drive_downloadable_version,
+      hasSupplementaryAssetObject: has_supplementary_asset_object,
+      isLegacy: is_legacy,
+      copyrightContent: currentLesson?.lesson_data.copyright_content,
+    };
+
+    const downloads = constructDownloadsArray(downloadsData);
+
+    const unitLessonsArray = constructLessonListingObjectArray(
+      res.unit_lessons as SyntheticUnitvariantLessons[],
+    );
+    const nextLessons = getNextLessonsInUnit(unitLessonsArray, args.lessonSlug);
```

Comment:
suggest refactoring so that `getNextLessonsInUnit` constructs the `unitLessonsArray` internally

Reply:
Refactored so downloads are constructed from different functions